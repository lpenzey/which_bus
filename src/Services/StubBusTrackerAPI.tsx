import sampleResponse from './__fixtures__/searchResponse.json';

const StubBusTrackerAPI = {
    requestTimeEstimate: async (route: number, stpid: number, format: string) => {
        return sampleResponse;
    }
};

export default StubBusTrackerAPI;