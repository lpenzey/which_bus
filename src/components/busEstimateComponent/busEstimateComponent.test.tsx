import React from 'react';
import BusEstimateComponent from './busEstimateComponent';
import { mount } from 'enzyme';
import Enzyme from 'enzyme';
import StubBusTrackerAPI from '../../Services/StubBusTrackerAPI'
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';

function flushPromises() {
    return new Promise(resolve => setImmediate(resolve));
}

describe('<BusEstimateComponent />', () => {
    Enzyme.configure({ adapter: new Adapter() })
    it('does a snapshot', () => {
        const wrapper = mount(
            <BusEstimateComponent api={StubBusTrackerAPI} />
        );

        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should update text when clicking on button', () => {
        const wrapper = mount(
            <BusEstimateComponent api={StubBusTrackerAPI} />
        );

        const button = wrapper.find('#estimateButton').at(0);
        button.simulate('click');

        return flushPromises().then(() => {
            wrapper.update()
            expect(toJSON(wrapper)).toMatchSnapshot();
        });
    });
});
