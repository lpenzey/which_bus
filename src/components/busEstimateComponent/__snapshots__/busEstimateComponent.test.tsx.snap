// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusEstimateComponent /> does a snapshot 1`] = `
<BusEstimateComponent
  api={
    Object {
      "requestTimeEstimate": [Function],
    }
  }
>
  <div
    className="timeTable"
  >
    ROUTE:  
    <br />
    STOP: 
    <br />
    DIRECTION: 
    <br />
    NEXT BUS ARRIVES AT: 
    <br />
    <Button
      active={false}
      disabled={false}
      id="estimateButton"
      onClick={[Function]}
      type="button"
      variant="primary"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        id="estimateButton"
        onClick={[Function]}
        type="button"
      >
        Click to Get Estimate
      </button>
    </Button>
  </div>
</BusEstimateComponent>
`;

exports[`<BusEstimateComponent /> should update text when clicking on button 1`] = `
<BusEstimateComponent
  api={
    Object {
      "requestTimeEstimate": [Function],
    }
  }
>
  <div
    className="timeTable"
  >
    ROUTE:  
    20
    <br />
    STOP: 
    Madison & Jefferson
    <br />
    DIRECTION: 
    Westbound
    <br />
    NEXT BUS ARRIVES AT: 
    12:02
    <br />
    <Button
      active={false}
      disabled={false}
      id="estimateButton"
      onClick={[Function]}
      type="button"
      variant="primary"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        id="estimateButton"
        onClick={[Function]}
        type="button"
      >
        Click to Get Estimate
      </button>
    </Button>
  </div>
</BusEstimateComponent>
`;
